worker/provisioner/provisioner_task.go

```diff
@@ -782,6 +782,12 @@ func (task *provisionerTask) constructInstanceConfig(
 		return nil, errors.Trace(err)
 	}
 
+	instanceConfig.Controller = &instancecfg.ControllerConfig{}
```

Comment:
I think this is ok to move here.
I can't think of a case where `len(pInfo.Jobs) > 0` will not be true - a node normally has at the least the HostUnits job defined. It would be interesting to explore if we need the jobs check at all.

Reply:
Yeah, you're right. It's probably not necessary here, unless some code downstream is checking `jobs == nil` vs `len(jobs) == 0` and making a distinction. Going to leave as is for now just in case.